misk-redis/src/testFixtures/kotlin/misk/redis/testing/FakeRedis.kt

```diff
@@ -411,4 +421,89 @@ class FakeRedis @Inject constructor(
   override fun publish(channel: String, message: String) {
     throw NotImplementedError("Fake client not implemented for this operation")
   }
+
+  override fun zadd(
+    key: String,
+    score: Double,
+    member: ByteString,
+    options: Set<String>
+  ): Long {
+    verifyOptions(options)
+    var newFieldCount = 0L
+    var elementsChanged = 0L
+    val optionsUpper = options.map { it.uppercase() }.toMutableList()
+    if (optionsUpper.isEmpty() || (optionsUpper.size == 1 && optionsUpper.contains(CH.name))) {
+      optionsUpper.add(XX.name)
+      optionsUpper.add(NX.name)
```

Comment:
Aren't these two options mutually exclusive?

Reply:
No. NX, LT, GT are mutually exclusive per the doc. 
In the absence of any options it behaves as though it has NX and XX.

Reply:
NX = Only add new elements. Don't update already existing elements.
XX = Only update elements that already exist. Don't add new elements.

So zadd is a no-op if it has both these options. Check out the command in `redis-cli`. It's not possible to specify both:
![Screenshot 2023-10-19 at 3 21 55 PM](https://github.com/cashapp/misk/assets/588701/69b762f3-c596-49a6-86bc-143132b4345f)

Reply:
the default behavior (without any options) is to add new elements and update existing. So it makes sense that from the UX p.o.v. that both NX and XX are not allowed together. I'll add it to the verification logic. 
For the implementation, I think it makes sense to treat them (NX, XX) as separate operations and do them together when no options are specified or just do the specified one. 
wdyt?

Reply:
> For the implementation, I think it makes sense to treat them (NX, XX) as separate operations and do them together when no options are specified or just do the specified one.

If by "do them together when no options are specified" you mean do an upsert, that is correct. zadd's default is to do an upsert

Reply:
Yes!